server/src/main/java/org/opensearch/index/query/RegexpQueryBuilder.java

```diff
@@ -299,7 +299,8 @@ protected Query doToQuery(QueryShardContext context) throws QueryShardException,
         int matchFlagsValue = caseInsensitive ? RegExp.ASCII_CASE_INSENSITIVE : 0;
         Query query = null;
         // For BWC we mask irrelevant bits (RegExp changed ALL from 0xffff to 0xff)
-        int sanitisedSyntaxFlag = syntaxFlagsValue & RegExp.ALL;
+        // The hexadecimal for DEPRECATED_COMPLEMENT is 0x10000. The OR condition ensures COMPLEMENT ~ is preserved
+        int sanitisedSyntaxFlag = syntaxFlagsValue & (RegExp.ALL | RegExp.DEPRECATED_COMPLEMENT);
```

Comment:
Is there any other condition we can rely upon? DEPRECATED_COMPLEMENT is marked for deletion in Lucene 11: https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/util/automaton/RegExp.java#L393-L399

Reply:
From https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/util/automaton/RegExp.java#L1343-L1346
We should consider that as breaking change with Lucene 11 and suggest some alternatives for users, example https://github.com/opensearch-project/OpenSearch/issues/18397#issuecomment-3016185863. We should even update todays docs https://docs.opensearch.org/docs/latest/query-dsl/regex-syntax/ to announce this.

Reply:
Do we need to use the deprecation logger to log a warning that this functionality will be removed in the next major version when we adopt Lucene 11?